mne_bids/tests/test_read.py

```diff
@@ -317,6 +317,37 @@ def test_handle_coords_reading():
     with pytest.raises(RuntimeError, match='Channels do not correspond'):
         raw_test = read_raw_bids(bids_fname, bids_root)
 
+    # make sure montage is not set if there are coordinates w/ 'n/a'
+    # but not set in raw.info['bads']
+    raw.info['bads'] = []
+    write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
+    electrodes_dict = _from_tsv(electrodes_fname)
+    for axis in ['x', 'y', 'z']:
+        electrodes_dict[axis][0] = 'n/a'
+        electrodes_dict[axis][3] = 'n/a'
+    _to_tsv(electrodes_dict, electrodes_fname)
+
+    # set channel status as 'good' explicitly
+    channels_fname = _find_matching_sidecar(bids_fname, bids_root,
+                                            "channels.tsv",
+                                            allow_fail=True)
+    channels_dict = _from_tsv(channels_fname)
+    channels_dict['status'][0] = 'good'
+    channels_dict['status'][3] = 'good'
+    raw_test = read_raw_bids(bids_fname, bids_root)
+    assert raw_test.info['dig'] is not None
+
+    # make sure montage is set if channel is bad
+    # set channel status as 'bad' explicitly
+    channels_fname = _find_matching_sidecar(bids_fname, bids_root,
+                                            "channels.tsv",
+                                            allow_fail=True)
+    channels_dict = _from_tsv(channels_fname)
+    channels_dict['status'][0] = 'bad'
+    channels_dict['status'][3] = 'bad'
+    raw_test = read_raw_bids(bids_fname, bids_root)
+    assert raw_test.info['dig'] is not None
```

Comment:
given the current code, it seems you should do a loop. But I'm missing something so far ...

Reply:
What would you do a loop over? All the raw.info['dig']?